### PR TITLE
34731 barebones theme port

### PR DESCRIPTION
**Pull request by JohnAlbin**

### CODE REVIEW

**Comment by JohnAlbin** [comment]
After adding `style="width: 100%"` to the logo.svg in my browser's HTML inspector (temporary work-around until the header ticket is completed), this is the screenshot from the PR env:

screen shot 2015-09-12 at 03 54 31

Pretty good considering I just converted page.html.twig, no?

**Comment by jibran** [comment]
Yeah looks good. Does this mean aGov is not going to use zen in D8? You mentioned <a href="https://www.drupal.org/node/2052399#comment-10235909">here</a> that zen was needed for the Drupal 8 port of aGov.

File: themes/agov/agov_whitlam/agov_whitlam.info.yml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+name: aGov Whitlam
+description: 'Drupal 8 aGov Theme.'
+screenshot: screenshot.jpg
```
Comment (jibran) [inline themes/agov/agov_whitlam/agov_whitlam.info.yml]: Can we convert this to svg?

File: themes/agov/agov_whitlam/templates/page.html.twig [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+  {% if page.sidebar_first and page.sidebar_second %}
+    {% set content_class =        'layout-3col__right-content' %}
+    {% set sidebar_first_class =  'layout-3col__left-sidebar' %}
+    {% set sidebar_second_class = 'layout-3col__left-sidebar' %}
+  {% elseif page.sidebar_second %}
+    {% set content_class =        'layout-3col__left-content' %}
+    {% set sidebar_second_class = 'layout-3col__right-sidebar' %}
+  {% elseif page.sidebar_first %}
+    {% set content_class =        'layout-3col__right-content' %}
+    {% set sidebar_first_class =  'layout-3col__left-sidebar' %}
```
Comment (jibran) [inline themes/agov/agov_whitlam/templates/page.html.twig]: Is this some kind of coding standard alignment of classes?

File: themes/agov/agov_whitlam/templates/page.html.twig [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      </div>{# ./header__name-and-slogan #}
+    {% endif %}
+
+    {{ page.secondary_menu }}
+
+    {{ page.header }}
+  </div>
+</header>
+
+<div class="navbar__wrapper">
```
Comment (jibran) [inline themes/agov/agov_whitlam/templates/page.html.twig]: Can this also be `<nav>` just like action links?

File: themes/agov/agov_whitlam/templates/page.html.twig [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    {% set sidebar_second_class = 'layout-3col__left-sidebar' %}
+  {% elseif page.sidebar_second %}
+    {% set content_class =        'layout-3col__left-content' %}
+    {% set sidebar_second_class = 'layout-3col__right-sidebar' %}
+  {% elseif page.sidebar_first %}
+    {% set content_class =        'layout-3col__right-content' %}
+    {% set sidebar_first_class =  'layout-3col__left-sidebar' %}
+  {% endif %}
+
+  <div class="{{ content_class }}">
```
Comment (jibran) [inline themes/agov/agov_whitlam/templates/page.html.twig]: Can we use `<section>` and `<main>` tags here?

**Comment by rcaracaus** [comment]
We are not QA'ing or reviewing anything in page.tpl.php as part of this PR, for now it is fine for it to use core's. When we actually cover the individual components in other tickets, we can address concerns around how it is structured.

I have already done quite a few things of the things like logo, jquery, and removed the template.php in the header PR  https://github.com/previousnext/agov/pull/650, I should have removed template.php in this PR I guess.

File: themes/agov/agov_whitlam/agov_whitlam.info.yml (hunk 1 above)
Comment (rcaracaus) [inline themes/agov/agov_whitlam/agov_whitlam.info.yml]: the screenshot should be jpg

File: themes/agov/agov_whitlam/templates/page.html.twig (hunk 2 above)
Comment (rcaracaus) [inline themes/agov/agov_whitlam/templates/page.html.twig]: we should probably use twig extends for the different layouts, but we don't need to adderss this for this basic PR

File: themes/agov/agov_whitlam/templates/page.html.twig (hunk 4 above)
Comment (rcaracaus) [inline themes/agov/agov_whitlam/templates/page.html.twig]: "HTML5 structure is a mess: we should probably ignore HTML structural elements altogether. They’ve been a bit of a disaster — we have essentially forked the spec, created plenty of broken outlines, and wasted too much time already trying to get our heads around a fundamentally broken system. Long live divs."

http://www.webdesignerdepot.com/2013/01/the-harsh-truth-about-html5s-structural-semantics-part-3/

File: themes/agov/agov_whitlam/templates/page.html.twig (hunk 2 above)
Comment (JohnAlbin) [inline themes/agov/agov_whitlam/templates/page.html.twig]: > Is this some kind of coding standard alignment of classes?

This is just a straight port of the PHP to Twig syntax. I found it slightly easier to read the classes when I aligned them, but its not a coding standard.

> we should probably use twig extends for the different layouts

Twig extends means we would have all of our variables duplicated in each of the layouts. I'm not in favor of that. All we need is 3 variables with the classes in them, which we have.

File: themes/agov/agov_whitlam/templates/page.html.twig (hunk 3 above)
Comment (JohnAlbin) [inline themes/agov/agov_whitlam/templates/page.html.twig]: There's no `<nav>` in Classy's version of page.html.twig. See line 62 of https://api.drupal.org/api/drupal/core%21themes%21classy%21templates%21layout%21page.html.twig/8

Pretty sure the `<nav>` is part of the page.primary_menu variable.

File: themes/agov/agov_whitlam/templates/page.html.twig (hunk 4 above)
Comment (JohnAlbin) [inline themes/agov/agov_whitlam/templates/page.html.twig]: > Can we use <section> and <main> tags here?

I compared this page.html.twig to Classy's and you are right; I missed the `<main>` element. But there's no sections in Classy's page and I don't see a reason to add one to our page; we have `<aside>` for our sidebars, `<header>` in our header, and (after the footer is ported), `<footer>` in our footer.

File: themes/agov/agov_whitlam/agov_whitlam.info.yml (hunk 1 above)
Comment (JohnAlbin) [inline themes/agov/agov_whitlam/agov_whitlam.info.yml]: > the screenshot should be jpg

All of core's themes use screenshot.png, but, unlike the logo.svg, the "screenshot" filename is configurable in the .info.yml. So I'm fine with it being a jpeg.

**Comment by JohnAlbin** [comment]
> Yeah looks good. Does this mean aGov is not going to use zen in D8? You mentioned here that zen was needed for the Drupal 8 port of aGov.

@jibran Yeah, Rob was tasked this week to port the theme to D8 instead of waiting for Zen.

**Comment by JohnAlbin** [comment]
> I have already done quite a few things of the things like logo, jquery, and removed the template.php in the header PR #650, I should have removed template.php in this PR.

No, you did remove it. And I explicitly added it back because it hasn't been ported yet. Don't delete things until they have been ported. Please read https://redmine.previousnext.com.au/issues/34731#note-6 carefully as it is how you should be approaching this Drupal 8 port.

**Comment by JohnAlbin** [comment]
> What is point of having svg as embedded PNG rather than just using a png? I have implemented a png in the header PR.

Unfortunately, Drupal has a hard-coded name for its default logo. In D7, you _had_ to name it logo.png or the "Use the default logo supplied by the theme" checkbox in the theme settings UI will be broken and you will see a broken image when the checkbox is checked.

In D8, the hard-coded name is logo.svg. Your header PR just unchecks that UI checkbox when you install. But a user can still check that box in the UI and will get a broken image. That's why we _must_ use logo.svg in agov_whitlam.

Why is it a "PNG embedded in an SVG" in this PR? Because I couldn't find a proper illustration-based SVG for the logo and I wanted to get this PR completed before I go on vacation in 2 hours. You will need to fix that SVG in the header PR.

File: themes/agov/agov_whitlam/templates/page.html.twig (hunk 4 above)
Comment (JohnAlbin) [inline themes/agov/agov_whitlam/templates/page.html.twig]: > Based on Luke Steven's article, I propose we use aria roles, but ignore semantic elements altogether.

Definitely, ARIA roles are **the most important.** 

The article's primary rationale for not using HTML5 tags was: *it's too hard to understand how to do it right*. But we are a smart group, not a single web developer making mistakes about how to read the HTML5 spec. So we can use proper HTML5 tags; Drupal 8 core can be one of our guides.

**Comment by JohnAlbin** [comment]
> We are not QA'ing or reviewing anything in page.tpl.php as part of this PR, for now it is fine for it to use core's.

I'm not sure why you are saying that when this PR _does_ have a page.html.twig and we are _not_ using core's.

If you want to defer the decision to change this PR until a later PR, that's fine. But page.html.twig is part of the bare bones port, IMO.

**Comment by JohnAlbin** [comment]
I have a couple new commits. Adding them now.

**Comment by rcaracaus** [comment]
Thanks for your thorough review and explanations!

**Comment by JohnAlbin** [comment]
Additional commits were:
- Add missing <main> element to page.html.twig.	d533031
- Replace excessive spacing with twig comments.	7cda182
- Accessibility fix: move HTML from region--footer-bottom to page so "back to top" link always appears.
- Add regions to .info.yml.	f2227e9
- Port region--footer to twig.	3a870af
- Remove Zen README from theme.	80ad7c5
- Add logo config to theme and fix indentation of features yaml.	9f3a8d4
- Remove the bits of the d7 .info file that have already been ported.

I realized the regions were broken because we have custom regions in our page, but I hadn't configured them in the .info.yml.

I also fixed up the twig syntax that @jibran mentioned.

**Pull request merged**